release/k3s/images.go

```diff
@@ -0,0 +1,169 @@
+package k3s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	reg "github.com/rancher/ecm-distro-tools/registry"
+)
+
+type RegistryClient interface {
+	Image(ctx context.Context, ref name.Reference) (reg.Image, error)
+}
+
+const (
+	K3sImagesFile = "k3s-images.txt"
```

Comment:
Does this need to be consumed outside of this package? This goes for this symbol, and all other types, interfaces,  functions, and methods.